pkg/compiler/codegen.go

```diff
@@ -225,8 +226,12 @@ func getBaseOpcode(t varType) (opcode.Opcode, opcode.Opcode) {
 
 // emitLoadVar loads specified variable to the evaluation stack.
 func (c *codegen) emitLoadVar(pkg string, name string) {
-	t, i := c.getVarIndex(pkg, name)
-	c.emitLoadByIndex(t, i)
+	vi := c.getVarIndex(pkg, name)
+	if vi.tv.Value != nil {
```

Comment:
And `c.getVarIndex()` can return a `nil`, why don't we check for it here?

Reply:
`codegen.getVarIndex` never returns nil, it creates a new local if nothing was found.

Reply:
Well it does invoke `c.scope.vars.getVarInfo(name)` and that can return a `nil`, isn't it?

Reply:
For the first call there is a `nil` check.
For the second call, the variable for `name` is created on a previous line.

May be we should restructure this to make it more obvious.